tests/run-pass/file_read.rs

```diff
@@ -2,12 +2,24 @@
 // compile-flags: -Zmiri-disable-isolation
 
 use std::fs::File;
-use std::io::Read;
+use std::io::{Read, Write};
 
 fn main() {
-    // FIXME: create the file and delete it when `rm` is implemented.
-    let mut file = File::open("./tests/hello.txt").unwrap();
-    let mut contents = String::new();
-    file.read_to_string(&mut contents).unwrap();
-    assert_eq!("Hello, World!\n", contents);
+    // FIXME: remove the file and delete it when `rm` is implemented.
+    let path = "./tests/hello.txt";
+    let bytes = b"Hello, World!\n";
+    // Test creating, writing and closing a file (closing is tested when `file` is dropped).
+    let mut file = File::create(path).unwrap();
+    // Writing 0 bytes should not change the file contents.
```

Comment:
Isn't the main issue here that writing 0 bytes should not require the pointer to be valid?